pkg/operator/minio/controller.go

```diff
@@ -99,7 +101,7 @@ func (c *MinioController) makeMinioHeadlessService(name, namespace string, spec
 		},
 		Spec: v1.ServiceSpec{
 			Selector:  map[string]string{k8sutil.AppAttr: minioLabel},
-			Ports:     []v1.ServicePort{{Port: spec.Port}},
+			Ports:     []v1.ServicePort{{Port: spec.Network.Ports[0].Port}},
```

Comment:
Is there any (good) reason we want the user to change the port here?
Every (ignoring headless Services for a moment) Service has its own Cluster IP, so where is the point in letting the user choose the port if it doesn't really matter in the end?
The (user) reason "we want to replace an existing S3 but can't change the address without modifying the application code" is not a (good) reason, especially not in a cloud native environment such as Kubernetes.
(Same applies for Pods as they all have their own IP (+ netns))

Reply:
That's a good point @galexrt, I don't really know of a strong reason or scenario to be exposing the port that the services use internally.  A client or user would be connecting to the external service they get to define as seen in the example: https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/minio/object-store.yaml#L37-L47

maybe this internal port shouldn't be configurable at all, and we should always just use 9000 for it.